### ISSUE TITLE
Code blocks should not have the zenburn background when using pygments / coderay

### ISSUE THREAD

**Issue by wackywendell** [asciidoctor/asciidoctor-reveal.js#27]
Right now, when using `:source-highlighter: pygments`, the background color of the code blocks will still be the color of the `zenburn` background, and not the appropriate one from `pygments`.

This might be a reveal.js issue (https://github.com/hakimel/reveal.js/issues/1114), but I haven't quite figured it out.

**Comment by mojavelinux** [comment]
You are correct that this is an issue with this backend. We need to be checking which syntax highlighter is in use before adding the zenburn style to the HTML output.

**Comment by wackywendell** [comment]
Well... I may be over my head, but I'm not sure its quite that simple. I tried just removing the `<link href="[...]/zenburn.css">` line from the HTML output, and it still has that same color.

Looking at the Developer Tools > Inspector in Firefox lead me to this:

```css
.reveal pre code {
  display: block;
  padding: 5px;
  overflow: auto;
  max-height: 400px;
  word-wrap: normal;
  background: #3F3F3F;
  color: #DCDCDC; }
```

Which comes from [simple.css](https://cdnjs.cloudflare.com/ajax/libs/reveal.js/3.0.0//css/theme/simple.css) line 140.

It looks to me like the `background` and `color` attributes shouldn't be there (i.e., its a problem with `reveal.js`), or should be overridden when not using `highlight.js`. Overridden might make more sense, as that also covers "plain" code blocks where pygments / coderay wasn't run.

**Comment by mojavelinux** [comment]
> should be overridden when not using highlight.js

Yes, this is the second part to the change. We have to do a lot of this in the default Asciidoctor stylesheet, so it's the same routine here.

**Comment by obilodeau** [comment]
Is this still an issue? Can I get a more precise reproducible case with screenshots?

**Comment by ggrossetie** [comment]
Revisiting this decade-old one — good news, it looks resolved upstream.

I checked the current default theme this converter targets (reveal.js 6.0.1, `black` theme). The `.reveal pre code` rule no longer hardcodes any `background`/`color`:

```css
.reveal pre code {
  word-wrap: normal;
  max-height: 400px;
  padding: 5px;
  display: block;
  overflow: auto;
}
```

That's exactly the `background: #3F3F3F; color: #DCDCDC;` pair flagged above (from reveal.js 3.0.0's `simple.css` at the time) — it's gone from the modern theme CSS. There's also no `zenburn` reference anywhere in this project's own code (Ruby or JS converter, bundled assets), so nothing on our side was ever forcing that background either — this was purely a reveal.js theme issue, and reveal.js fixed it themselves at some point since.

Closing as resolved upstream — please reopen with a fresh repro (theme + syntax highlighter + reveal.js version) if you can still reproduce this on a current setup.
